### PR TITLE
MegaRAID Plugin: Add support of system_fw_version_get()

### PR DESCRIPTION
**Pull request by cathay4t**
```
* Include these version in returned string
    * "package build version":
      The version string user could compare to download firmware package.
    * BIOS version:
      The version of BIOS of this card.
    * FW version:
      Firmware version.
```

Signed-off-by: Gris Ge fge@redhat.com

### CODE REVIEW

**Comment by joehandzik** [comment]
Guys, I was explicitly told to not add it because it would break binary compatibility...that's originally where I had the code! Let me know if I'm misinterpreting what is being suggested. I totally agree that this should be collapsed into the base set of system information, it was real heavy to write over a hundred lines of template code to get the firmware string out.

File: plugin/megaraid/megaraid.py [hunk 1]
```diff
@@ -791,3 +792,17 @@ def volume_raid_create(self, name, raid_type, disks, strip_size,
...
+
+    @_handle_errors
+    def system_fw_version_get(self, system, flags=0):
+        if not system.plugin_data:
+            raise LsmError(
+                ErrorNumber.INVALID_ARGUMENT,
+                "Ilegal input system argument: missing plugin_data property")
+
+        output = self._storcli_exec([system.plugin_data, "show"])
+        return "Pacakge: %s, BIOS: %s, FW: %s" % (
```
Comment (joehandzik) [inline plugin/megaraid/megaraid.py]: Typo, should be "Package:

**Comment by joehandzik** [comment]
@tasleson I guess I'll just continue the conversation here. Now that I've actually thought about this a bit, I suppose I could have added a firmware element to the type in lsm_datatypes.hpp. But then there's a bunch of finagling I'd need to do around initialization and allocation (as far as I can tell, please correct me if I'm wrong) to get something up and stumbling. What I have here (I speculate) is an easier implementation to read and understand until we're ok with breaking binary compatibility in 2.0. Thoughts?

**Comment by cathay4t** [comment]
@joehandzik This pull request is just add the same function `system_fw_version_get()`
to MegaRAID plugin.
May I know which part of this pull request indicate I broke 1.x compatibility?

The binary compatibility we are maintaining is to make sure:
1. Old plugin can run on new client API without any re-compile or change.
        This make sure third party plugin still works. But does not stop us from updating
        our open source plugins.
2. Old client API can run on new plugin without any re-compile or change.
3. Old plugin or old client can run on new daemon without any re-compile or change.

**Comment by cathay4t** [comment]
New patch uploaded with rebase and typo fix.

**Comment by joehandzik** [comment]
Oh, sorry if I wasn't clear, @cathay4t, I was saying that my previous PR (https://github.com/libstorage/libstoragemgmt/pull/43) broke binary compatibility, not that the change you've added here does.

**Pull request closed without merging**